sno/gpkg_adapter.py

```diff
@@ -203,11 +203,9 @@ def _gpkg_srs_id(v2_obj):
 
 def wkt_to_gpkg_spatial_ref_sys(wkt):
     """Given a WKT crs definition, generate a gpkg_spatial_ref_sys meta item."""
-    # TODO: Better support for custom WKT. https://github.com/koordinates/sno/issues/148
     spatial_ref = SpatialReference(wkt)
-    spatial_ref.AutoIdentifyEPSG()
```

Comment:
It's pretty common to have WKT _without_ authority codes, but otherwise identical to an EPSG definition. This is particularly common with classic ESRI tools. If the parameters exactly match EPSG:1234 then we should use that for it's CRS identifier and that's what `AutoIdentifyEPSG` tries to do.

eg. WKT2 for NZTM (EPSG:2193):
```PROJCRS["NZGD2000 / New Zealand Transverse Mercator 2000",
    BASEGEOGCRS["NZGD2000",
        DATUM["New Zealand Geodetic Datum 2000",
            ELLIPSOID["GRS 1980",6378137,298.257222101,
                LENGTHUNIT["metre",1]]],
        PRIMEM["Greenwich",0,
            ANGLEUNIT["degree",0.0174532925199433]],
        ID["EPSG",4167]],
    CONVERSION["New Zealand Transverse Mercator 2000",
        METHOD["Transverse Mercator",
            ID["EPSG",9807]],
        PARAMETER["Latitude of natural origin",0,
            ANGLEUNIT["degree",0.0174532925199433],
            ID["EPSG",8801]],
        PARAMETER["Longitude of natural origin",173,
            ANGLEUNIT["degree",0.0174532925199433],
            ID["EPSG",8802]],
        PARAMETER["Scale factor at natural origin",0.9996,
            SCALEUNIT["unity",1],
            ID["EPSG",8805]],
        PARAMETER["False easting",1600000,
            LENGTHUNIT["metre",1],
            ID["EPSG",8806]],
        PARAMETER["False northing",10000000,
            LENGTHUNIT["metre",1],
            ID["EPSG",8807]]],
    CS[Cartesian,2],
        AXIS["northing (N)",north,
            ORDER[1],
            LENGTHUNIT["metre",1]],
        AXIS["easting (E)",east,
            ORDER[2],
            LENGTHUNIT["metre",1]],
    USAGE[
        SCOPE["unknown"],
        AREA["New Zealand - onshore"],
        BBOX[-47.33,166.37,-34.1,178.63]],
    ID["EPSG",2193]]
```

ESRI WKT for the same CRS:
```
PROJCS["NZGD_2000_New_Zealand_Transverse_Mercator",GEOGCS["GCS_NZGD_2000",DATUM["D_NZGD_2000",SPHEROID["GRS_1980",6378137.0,298.257222101]],PRIMEM["Greenwich",0.0],UNIT["Degree",0.0174532925199433]],PROJECTION["Transverse_Mercator"],PARAMETER["False_Easting",1600000.0],PARAMETER["False_Northing",10000000.0],PARAMETER["Central_Meridian",173.0],PARAMETER["Scale_Factor",0.9996],PARAMETER["Latitude_Of_Origin",0.0],UNIT["Meter",1.0]]
```

Reply:
Ok. I'll investigate more and re-add AutoIdentifyEPSG.

I might end up calling it and checking the result to see if it did what we wanted, if not keep the original.

Reply:
I think maybe call `AutoIdentifyEPSG()` and use the auth name+code for the identifier if it returns one. Then keep the original WKT?

Reply:
the answer here seems to be that AutoIdentifyEPSG just ... doesn't work. 

For both of those inputs you gave us above (if you strip off the ID from the WKT2 so it's not trivial), `AutoIdentifyEPSG` raises a `RuntimeError`...

One way to get something useful is to call `FindMatches` instead:

```python
>>> s.FindMatches()[0]
(<osgeo.osr.SpatialReference; proxy of <Swig Object of type 'OSRSpatialReferenceShadow *' at 0x10e9f06c0> >, 100)
>>> s.FindMatches()[0][0].GetAuthorityCode(None)
'2193'
```

This seems to work for me for both inputs, maybe we should just use that instead?

Reply:
With the ESRI example, calling `AutoIdentifyEPSG()` _then_ `FindMatches()` reduces the FindMatches confidence to 25% (it's 100 if you don't call `AutoIdentifyEPSG` first)

Based on this it seems to me that @olsen232 is correct that `AutoIdentifyEPSG` is destructive and fairly useless and should be avoided